Hermes/appscale/hermes/stats/subscribers/profile.py

```diff
@@ -47,22 +53,196 @@ def receive(self, nodes_stats_dict):
           writer.writerow(row)
 
   def _prepare_file(self, node_ip):
+    """ Prepares CSV file with name <node-IP>.csv for appending new lines.
+
+    Args:
+      node_ip: a string representation of node IP
+    Returns:
+      file object opened for appending new data
+    """
     file_name = path.join(self._directory, '{}.csv'.format(node_ip))
     if not path.isfile(file_name):
+      # Create file and write header
       with open(file_name, 'w') as csv_file:
         csv.writer(csv_file).writerow(self._header)
+    # Open table for appending data
     return open(file_name, 'a')
 
 
 class ClusterProcessesProfileLog(StatsSubscriber):
 
+  @attr.s(cmp=False, hash=False, slots=True)
+  class ServiceSummary(object):
+    """
+    This data structure is kind of service summary accumulator.
+    When new stats is received, ServiceSummary is created for each service
+    and then cpu time and memory usage of each process running this service
+    is added to the summary.
+    """
+    cpu_time = attr.ib(default=0)
+    resident_mem = attr.ib(default=0)
+    unique_mem = attr.ib(default=0)
+
   def __init__(self, include_lists=None):
-    # TODO
-    pass
+    """ Initializes profile log for cluster processes stats.
+    Renders header according to include_lists in advance and
+    creates base directory for processes stats profile log.
+    It also reads header of summary file (if it exists) to identify
+    order of columns.
+
+    Args:
+      include_lists: an instance of IncludeLists describing which fields
+          of processes stats should be written to CSV log.
+    """
+    self._header = ['utc_timestamp'] + converter.get_stats_header(
+      process_stats.ProcessStats, include_lists)
+    self._include_lists = include_lists
+    self._directory = path.join(PROFILE_LOG_DIR, 'processes')
+    self._summary_file_name_template = 'summary-{resource}.csv'
+    try:
+      makedirs(self._directory)
+    except OSError as os_error:
+      if os_error.errno == errno.EEXIST and path.isdir(self._directory):
+        pass
+      else:
+        raise
+    self._summary_columns = self._get_summary_columns()
 
   def receive(self, processes_stats_dict):
-    # TODO
-    pass
+    """ Implements receive method of base class. Saves newly produced
+    cluster processes stats to a list of CSV files.
+    One detailed file for each process on every node and 3 summary files.
+
+    Args:
+      processes_stats_dict: a dict with node IP as key and list of
+          ProcessesStatsSnapshot as value
```

Comment:
I couldn't find any best practices for multiline argument descriptions, but I think that adding 4 spaces before the second and further lines is readable enough. I added a period in the end of multiline descriptions. Could you suggest a better solution?